### ISSUE TITLE
bug(gnovm): incorrect type comparison

### ISSUE THREAD

**Issue by moul** [gnolang/gno#2386]
![CleanShot 2024-06-18 at 16 59 30](https://github.com/gnolang/gno/assets/94029/908f6d35-88b9-4aa6-a3de-eecdbd1c277c)

https://go.dev/play/p/-y5gRjR9erA
https://play.gno.land/p/B9WLR_AsqYY

---

In GnoVM, type aliases with the same value are considered equal, while in Go, they are considered different without explicit casting. This leads to inconsistent behavior between Gno and Go.

## Example

The following code returns `false, false, false` in Go, but `true, false, false` in Gno:

```go
package main

import "fmt"

func check(v1, v2 interface{}) bool {
	return v1 == v2
}

func main() {
	type t1 int
	type t2 int
	v1 := t1(1)
	v2 := t2(1)
	v3 := t2(3)

	fmt.Println("v1, v2", v1, v2, check(v1, v2))
	fmt.Println("v1, v3", v1, v3, check(v1, v3))
	fmt.Println("v2, v3", v2, v3, check(v2, v3))
}
```

**Comment by deelawn** [comment]
This is not fixed by #1426

**Comment by ltzmaxwell** [comment]
this should be fixed in #1890.
similar context with https://github.com/gnolang/gno/issues/1376.

**Comment by ltzmaxwell** [comment]
this is fixed in https://github.com/gnolang/gno/pull/1890, please see:
https://github.com/gnolang/gno/blob/de5ef0021f4126c62c3caa0654fca2b15f060b2c/gnovm/tests/files/types/cmp_iface_7.gno
